### PR TITLE
doc: add custom domain instruction

### PR DESCRIPTION
**Pull request by michaellzc**
part of https://github.com/sourcegraph/pr-faqs/issues/126

this is not implemented yet, but adding user-facing docs first to set the right expectation.

## Test plan

docs only

### CODE REVIEW

**Comment by michaellzc** [comment]
Current dependencies on/for this PR:
* main
  * **PR #50622** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/50622" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/50622?utm_source=stack-comment).

**Review by unknwon — APPROVED** [review]

**Review by daxmc99 — APPROVED** [review]

File: doc/cloud/index.md [hunk 1]
```diff
@@ -65,6 +65,39 @@ All Sourcegraph features are avilable on Sourcegraph Cloud instances out-of-the-
 - Automatic monthly [upgrades](../admin/updates/index.md) and maintenance.
 - Regular reassessment of resource utilization based on your organization's unique usage to determine if costs can be reduced without impact to service. Additionally, you will automatically benefit from any committed use cloud provider discounts we receive.
 
+### Custom domain
```
Comment (bobheadxi) [inline doc/cloud/index.md:68]:
```suggestion
### Custom domains
```

File: doc/cloud/index.md [hunk 2]
```diff
@@ -65,6 +65,39 @@ All Sourcegraph features are avilable on Sourcegraph Cloud instances out-of-the-
...
+
+Sourcegraph Cloud provides all customer instances a `customer.sourcegraph.com` domain. This domain is fully managed by Sourcegraph, including DNS and HTTPS.
```
Comment (bobheadxi) [inline doc/cloud/index.md:70]: Wont this be `customer.sourcegraphcloud.com`?

File: doc/cloud/index.md [hunk 3]
```diff
@@ -65,6 +65,39 @@ All Sourcegraph features are avilable on Sourcegraph Cloud instances out-of-the-
...
+However, to provide better branding and a more seamless experience for your users, you may bring your own company domain.
```
Comment (bobheadxi) [inline doc/cloud/index.md:71]:
```suggestion
However, to provide better branding and a more seamless experience for your users, you may bring your own company domain, for example `sourcegraph.company.io`.
```

File: doc/cloud/index.md [hunk 4]
```diff
@@ -65,6 +65,39 @@ All Sourcegraph features are avilable on Sourcegraph Cloud instances out-of-the-
...
+
+In order to use your own domain, you need to perform an one-time setup by adding DNS records at your authoritative DNS. These DNS records are neccessary to ensure that your users can access your Sourcegraph instance via the custom domain, and also to ensure we can provide managed TLS certificates for your instance. See a [list of DNS records to be created by your organization](#list-of-dns-records-to-be-created-by-your-organization) below as an example. Additionally, your custom domain's [CAA records](https://blog.cloudflare.com/caa-of-the-wild/) should permit our upstream certificate authorities to issue certificates for your domain, follow the [instructions](#verify-caa-records) below to verify your CAA records.
+
+Please reach out to your Sourcegraph account team to request a custom domain to be configured for your Sourcegraph Cloud instance.
+
+#### List of DNS records to be created by your organization
```
Comment (bobheadxi) [inline doc/cloud/index.md:77]:
```suggestion
#### DNS records to be created by your organization
```

File: doc/cloud/index.md [hunk 5]
```diff
@@ -40,7 +40,7 @@ As part of this service you will receive a number of benefits from our team, inc
 - Advising if managed instances are right for your organization.
 - Initial resource estimations based on your organization & code size.
 - Putting forward a transparent deployment & cost estimate plan.
-- Your own `example.sourcegraph.com` domain with fully managed [DNS & HTTPS](../admin/http_https_configuration.md).
+- Your own `example.sourcegraph.com` domain with fully managed [DNS & HTTPS](../admin/http_https_configuration.md). Optionally, you can [bring your own domain](#custom-domain).
```
Comment (bobheadxi) [inline doc/cloud/index.md:43]: Also here, I think the domain will be `example.sourcegraphcloud.com` going forward right?

**Review by bobheadxi — APPROVED** [review]
Nice 😁

**Pull request merged**